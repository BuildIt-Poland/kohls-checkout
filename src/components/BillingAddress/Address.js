import styled from 'styled-components';

const Address = styled.address`
  padding: 0;
  font-style: normal;
`;

export default Address;
