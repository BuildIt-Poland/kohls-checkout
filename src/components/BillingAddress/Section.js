import styled from 'styled-components';

import { spacingMedium, spacingLarge } from '../../styles/designTokens';

const Section = styled.section`
  margin-bottom: ${spacingMedium};

  &:last-child {
    margin-bottom: ${spacingLarge};
  }
`;

export default Section;
