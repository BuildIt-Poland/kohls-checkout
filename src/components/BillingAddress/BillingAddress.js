import React from 'react';
import PropTypes from 'prop-types';

import SectionHeader from '../SectionHeader';
import ModalTrigger from '../ModalTrigger';
import Address from './Address';
import Text from './Text';

function BillingAddress({ name, address }) {
  const modalTrigger = (
    <ModalTrigger title="Billing Address" content="">
      <Text underline>Open Modal</Text>
    </ModalTrigger>
  );
  return (
    <section>
      <SectionHeader actionElement={modalTrigger}>Billing Address</SectionHeader>
      <Address>
        <Text bold>{name}</Text>
        <Text>{address.street}</Text>
        <Text>{address.city}</Text>
        <Text>{address.phoneNumber}</Text>
      </Address>
    </section>
  );
}

BillingAddress.propTypes = {
  name: PropTypes.string.isRequired,
  address: PropTypes.object.isRequired
};

export default BillingAddress;
