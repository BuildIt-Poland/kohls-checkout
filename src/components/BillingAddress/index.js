import { connect } from 'react-redux';

import BillingAddress from './BillingAddress';

function mapStateToProps(state) {
  return {
    address: state.shippingAddress
  };
}

export default connect(mapStateToProps)(BillingAddress);
