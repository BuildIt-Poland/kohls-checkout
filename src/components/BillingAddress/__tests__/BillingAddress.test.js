import React from 'react';
import { create } from 'react-test-renderer';

import BillingAddress from '../BillingAddress';

describe('COMPONENT - BillingAddress', () => {
  it('renders BillingAddress component', () => {
    const component = create(<BillingAddress />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
