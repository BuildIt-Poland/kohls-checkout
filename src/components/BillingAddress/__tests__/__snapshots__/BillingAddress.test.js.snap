// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - BillingAddress renders BillingAddress component 1`] = `
Array [
  .c0 {
  font-size: 2.4rem;
  text-align: left;
  margin: 1rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 0;
  background: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 2rem;
  min-height: 2rem;
  cursor: pointer;
}

.c1 svg {
  display: block;
}

<h2
    className="c0"
  >
    Billing Address
    <button
      className="c1"
      data-testid="modal-trigger"
      onClick={[Function]}
    >
      <svg
        height="18"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="9"
          cy="9"
          fill="#222"
          r="9"
        />
        <path
          d="M9.833 6.241H8.167V4.724h1.666v1.517zM8.167 7.49h1.666v5.786H8.167V7.49z"
          fill="#fff"
        />
      </svg>
    </button>
  </h2>,
  .c0 {
  padding: 0;
  font-style: normal;
}

.c1 {
  font-size: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  line-height: 1.4;
  margin: 0;
  margin-bottom: 1rem;
  color: #222222;
  margin: 0;
}

.c1:last-child {
  margin-bottom: 0;
}

.c2 {
  font-size: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
  line-height: 1.4;
  margin: 0;
  margin-bottom: 1rem;
  color: #222222;
  margin: 0;
}

.c2:last-child {
  margin-bottom: 0;
}

<address
    className="c0"
  >
    <p
      className="c1"
    >
      John Robinson
    </p>
    <p
      className="c2"
    >
      228 Park Ave S
    </p>
    <p
      className="c2"
    >
      New York, NY 10003
    </p>
    <p
      className="c2"
    >
      (444) 444-4444
    </p>
  </address>,
]
`;
